ForumProject/communications/consumers.py

```diff
@@ -0,0 +1,110 @@
+import json
+
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import WebsocketConsumer
+
+from .models import Room, Message
+
+
+class ChatConsumer(WebsocketConsumer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(args, kwargs)
+        self.room_name = None
+        self.room_group_name = None
+        self.room = None
+        self.messages = None
+        self.user = None
+        self.user_inbox = None
+
+    def connect(self):
+        self.room_name = self.scope['url_route']['kwargs']['room_name']
+        self.room_group_name = f'chat_{self.room_name}'
+        self.room = Room.objects.get(name=self.room_name)
+        self.messages = Message.objects.filter(room=self.room)
```

Comment:
You are fetching messages, but these messages are only used if the user is authenticated. Consider moving this inside the authentication check block to avoid unnecessary database queries when the user is not authenticated.

Reply:
Done, thanks